pymeasure/instruments/mksinst/mks937b.py

```diff
@@ -0,0 +1,288 @@
+                 "unknown (24)": "24",  # manual does not specify
+                 "no gauge": "NG",
+                 "no connection": "NC"}
+
+_ion_gauge_status = {"Wait": "W",
+                     "Off": "O",
+                     "Protect": "P",
+                     "Degas": "D",
+                     "Control": "C",
+                     "Rear panel Ctrl off": "R",
+                     "HC filament fault": "H",
+                     "No gauge": "N",
+                     "Good": "G",
+                     "NOT_IONGAUGE": "NAK152",
+                     }
+
+
+class MKS937B(Instrument):
+    """ MKS 937B vacuum gauge controller
+
+    :param adapter: pyvisa resource name of the instrument or an instance of
+                    MKSSerialAdapter, or MKSVISAAdapter.
+    :param address: device address included in every message to the instrument
+                    (default=253)
+    :param kwargs: Any valid key-word argument for Instrument
+    """
+
+    def __init__(self, adapter, address=253, **kwargs):
+        kwargs.setdefault("preprocess_reply", self.extract_reply)
+        if isinstance(adapter, str):
+            adapter = MKSVISAAdapter(adapter, **kwargs)
+        super().__init__(
+            adapter,
+            "MKS 937B vacuum gauge controller",
+            includeSCPI=False,
+            **kwargs
+        )
+        self.address = address
+        # compiled regular expression for finding numerical values in reply strings
+        self._re_fullresponse = re.compile(fr"@{self.address:03d}(.*)")
+        self._re_response_value = re.compile(fr"@{self.address:03d}ACK(.*)")
+        # set address in commands via dynamic properties
+        for prop in ["serial", "pressure1", "pressure2", "pressure3",
+                     "pressure4", "pressure5", "pressure6", "all_pressures",
+                     "combined_pressure1", "combined_pressure2",
+                     "sensor_typeA", "sensor_typeB", "sensor_typeC",
+                     "ion_gauge_status1", "ion_gauge_status3",
+                     "ion_gauge_status5", "unit",
+                     "power_status1", "power_status2", "power_status3",
+                     "power_status4", "power_status5", "power_status6",
+                     ]:
+            setattr(self, f"{prop}_command_process", self.command_process)
+
+    def extract_reply(self, reply):
+        """ preprocess_reply function which tries to extract <Response> from
+        '@<aaa>ACK<Response>;FF'. If <Response> can not be identified the orignal string
+        is returned.
+        :param reply: reply string
+        :returns: string with only the response, or the original string
+        """
+        rvalue = self._re_response_value.search(reply)
+        if rvalue:
+            return rvalue.groups()[0]
+        else:
+            full_reply = self._re_fullresponse.search(reply)
+            if full_reply:
+                return full_reply.groups()[0]
+        return reply
```

Comment:
This method depends on self (see self._re_response_value, self.re_fullresponse) which need to be instance variables since they depend on the device address. They could be made generic regular expressions, but I find it best to check it with the actual device address and not just with any 3 digit number. If this device would be used on a real bus connection (RS485) it is likely best in this way.

Reply:
Sorry, I missed that self, so you're fine.